enterprise/frontend/src/embedding-sdk/components/private/InteractiveQuestion/components/Notebook.tsx

```diff
@@ -1,33 +1,53 @@
 import { useMemo } from "react";
 
-import { useInteractiveQuestionContext } from "embedding-sdk/components/public/InteractiveQuestion/context";
+import { useInteractiveQuestionContext } from "embedding-sdk/components/private/InteractiveQuestion/context";
 import { useSelector } from "metabase/lib/redux";
 import { default as QBNotebook } from "metabase/query_builder/components/notebook/Notebook";
 import {
   isQuestionDirty,
   isQuestionRunnable,
 } from "metabase/query_builder/utils/question";
+import { getMetadata } from "metabase/selectors/metadata";
 import { getSetting } from "metabase/selectors/settings";
 import { ScrollArea } from "metabase/ui";
+import { sourceTableOrCardId } from "metabase-lib";
+import type Question from "metabase-lib/v1/Question";
 
 type NotebookProps = { onApply?: () => void };
 
 export const Notebook = ({ onApply = () => {} }: NotebookProps) => {
   const { question, originalQuestion, updateQuestion, runQuestion } =
     useInteractiveQuestionContext();
+  const metadata = useSelector(getMetadata);
 
-  const isDirty = useMemo(() => {
-    return isQuestionDirty(question, originalQuestion);
-  }, [question, originalQuestion]);
+  const isDirty = useMemo(
+    () => isQuestionDirty(question, originalQuestion),
+    [question, originalQuestion],
+  );
 
-  const isRunnable = useMemo(() => {
-    return isQuestionRunnable(question, isDirty);
-  }, [question, isDirty]);
+  const isRunnable = useMemo(
+    () => isQuestionRunnable(question, isDirty),
+    [question, isDirty],
+  );
 
   const reportTimezone = useSelector(state =>
     getSetting(state, "report-timezone-long"),
   );
 
+  const handleUpdateQuestion = async (nextQuestion: Question) => {
+    const query = nextQuestion.query();
+    const sourceTableId = sourceTableOrCardId(query);
+    const table = metadata.table(sourceTableId);
+    const databaseId = table?.db_id;
+
+    const nextQuestionWithDatabaseId = nextQuestion.setDatasetQuery({
+      ...nextQuestion.datasetQuery(),
+      database: databaseId ?? null,
+    });
```

Comment:
:x: This is incorrect - queries should be manipulated only with MLv2 (`Question` is not part of MLv2).

See `handleChange` in `NotebookDataPicker` + `handleTableChange` in `DataStep` to see an example of how to change source table of a query (when you change the table, database will also change).

----

Why do we need to set `database` here?

Reply:
Exactly why we needed your help 😄 when we were just calling `updateQuestion` without this extra logic, the rest of the notebook steps and the visualization button weren't showing up - it would literally just show the selected data source and that's it. 

After comparing the Notebook component in the main app and the SDK, I noticed that the main difference was that database wasn't populated in the SDK question query, while in the main app it was. So I tried a few ways but this seemed to be the only way to get things to work in SDK land

Reply:
> I noticed that the main difference was that database wasn't populated in the SDK question query, while in the main app it was

We need to find out why it was missing.

Is table metadata for the data source present in redux store?
Notice that we're explicitly loading it when changing data source: https://github.com/metabase/metabase/blob/020f98c28c6da5f930516cc339bd04681778ce0a/frontend/src/metabase/query_builder/components/notebook/NotebookDataPicker/NotebookDataPicker.tsx#L68-L75

Reply:
If someone for whatever reason ever reads this thread, the solution to why the database wasn't in the redux store was because we don't explicitly load database metadata into the redux store in the Notebook component in the main app (and thus also doesn't load the data for the SDK). The main app operates with the assumption that all data is loaded - an assumption that the SDK team can't operate on.

All that is to say, the solution is to explicitly load the database data with the `useDatabaseListQuery` hook at the start of the component. With that data, the Notebook will operate as normal.